### DISCUSSION TITLE
mod_sharedutils_dialogChoices implementation

### DISCUSSION THREAD

**Discussion by prokar** [Aelto/tw3-shared-utils#17] [General]
I'm trying to implement dialogue, run SU_test from the console and nothing appears. Were there any changes in 4.04?

**Comment by Aelto** [comment]
Nothing that should break it as far as i'm aware. Out of the box the `example.ws` is not bundled in the scripts so you'd have to manually move the file from the `example` folder and put it in the local folder of the mod. Did you do it?

**Reply by prokar** [reply]
Thank you for your answer.
I tried in various ways and nothing happened.

**Reply by Aelto** [reply]
Okay so i just tested and it works as expected, however you must move the example file as i explained above:
![image](https://github.com/Aelto/tw3-shared-utils/assets/13685633/a4d322e4-4219-4b6d-9ff1-775760477427)


- Open `mod_sharedutils_dialogChoices` and select the `example` folder
- Go in the `content` folder near it, then `scripts/local/sharedutils` and place the `example` folder inside it.
- Once it is done you should have a `dialog_choice.ws` file and the `example` alongside it.
- Boot the game, let it compile the scripts, then run the `SU_test` command as shown in the screenshot above

**Reply by prokar** [reply]
Ok, I'll check it out, thanks for your time.  👍

**Reply by Aelto** [reply]
However i just noticed the synchronous function with the event listener isn't working properly, the choices are displayed but the GUI module isn't cleaned correctly and it doesn't close like it should or like it does with the latent function. If you can, use the latent function instead or wait until I fix it

**Reply by Aelto** [reply]
I just pushed two quick commits, the event based solution is now working as intended. So now with the latest changes, if you move the file to the right place everything should work exactly like shown in the video link from the module's README !

**Reply by prokar** [reply]
Sorry for delay.
Just one commit required, working well.

I have one more question, unrelated to the thread.
Is it possible to get the ID of SItemUniqueId object as a string or int value? .Default is a value of name type. Can convert name to string, but not string to name. I don't know the struct SItemUniqueId, can I get another unique Id of the int or string type from other fields?

**Reply by Aelto** [reply]
Good to hear it worked.

`name` to `string` conversion is possible but not the opposite, that's normal. These unique IDs are related to inventory items, and the CInventoryComponent class has tons of methods that accept either names to get unique IDs, or unique IDs to get names in return. The component is what you should look at if you wish to manipulate or convert these unique ids

**Reply by prokar** [reply]
Thank you :+1:
